presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -367,7 +367,7 @@ private PlanRoot doAnalyzeQuery()
         stateMachine.setOutput(output);
 
         // fragment the plan
-        SubPlan subplan = PlanFragmenter.createSubPlans(stateMachine.getSession(), metadata, plan);
+        SubPlan subplan = PlanFragmenter.createSubPlans(stateMachine.getSession(), metadata, splitManager, plan);
```

Comment:
I forget if we discussed this, but why isn't all the necessary information available from table metadata? If that's the case, the fragmenter shouldn't have to rely on SplitManager.